components/Checkout/Billing.component.jsx

```diff
@@ -1,5 +1,46 @@
 import { useForm } from 'react-hook-form';
 import { InputField } from '../Input/InputField.component';
+import { getCustomNumberValidation } from '../../functions/functions';
+
+const inputs = [
+  { label: 'Fornavn', name: 'firstName' },
+  { label: 'Etternavn', name: 'lastName' },
+  { label: 'Adresse', name: 'address1' },
+  {
+    label: 'Postnummer',
+    name: 'postcode',
+    customValidation: getCustomNumberValidation(
+      {
+        minLength: 'Postnummer må være minimum 4 tall',
+        maxLength: 'Postnummer må være maksimalt 4 tall',
+        pattern: 'Postnummer må bare være tall',
+      },
+      4
+    ),
+  },
+  { label: 'Sted', name: 'city' },
+  {
+    label: 'Epost',
+    name: 'email',
+    customValidation: getCustomNumberValidation(
+      { pattern: 'Du må oppgi en gyldig epost' },
+      undefined,
+      /[^@]+@[^@]+\.[^@]+/i
```

Comment:
Getting a code quality warning here.

Maybe take a look and see if it can be fixed?

The error I am seeing is "Make sure the regex used here, which is vulnerable to super-linear runtime due to backtracking, cannot lead to denial of service"

Reply:
@zaxoavoki

Reply:
https://www.abstractapi.com/guides/email-validation-regex-javascript

May be some useful tips here